cppwinrt/code_writers.h

```diff
@@ -1131,15 +1131,21 @@ namespace cppwinrt
                 // we intentionally ignore errors when unregistering event handlers to be consistent with event_revoker
                 format = R"(    template <typename D%> auto consume_%<D%>::%(%) const noexcept
     {%
-        WINRT_IMPL_SHIM(%)->%(%);%
+        const auto castedResult = static_cast<% const&>(static_cast<D const&>(*this));
+        const auto abiType = *(abi_t<%>**)&castedResult;
+        check_cast_result(abiType);
+        abiType->%(%);%
     }
 )";
             }
             else
             {
                 format = R"(    template <typename D%> auto consume_%<D%>::%(%) const noexcept
     {%
-        WINRT_VERIFY_(0, WINRT_IMPL_SHIM(%)->%(%));%
+        const auto castedResult = static_cast<% const&>(static_cast<D const&>(*this));
```

Comment:
I believe this creates a copy and refcount bump.
```suggestion
        auto const& castedResult = static_cast<% const&>(static_cast<D const&>(*this));
```

Reply:
Thanks for the fix 👍.

I did `const auto&` but let me know if you think `auto const&` more closely matches the style of this project.  I believe they are equally correct.

Reply:
cppwinrt uses east const.

Reply:
Updated to code gen east const.